src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -812,31 +808,55 @@ export default class RouteModel extends BaseModel {
-        const indexOfDivergentLeg = this._findIndexOfLegContainingWaypointName(waypointName);
+    /**
+     * Append a provided route model into this RouteModel
+     *
+     * This method only serves to call the method that contains the appropriate logic
+     * based on the type of leg in which the divergent waypoint resides, since this
+     * heavily weighs in to how the merging of the routes should be done.
+     *
+     * @for RouteModel
+     * @method _appendRouteModelBeginningAtWaypointName
+     * @param divergentWaypointName {string} name of waypoint at which the two routes have continuity
+     * @param routeModel {RouteModel} the RouteModel to be absorbed into this
+     * @return {array} [success of operation, readback]
+     */
+    _appendRouteModelBeginningAtWaypointName(divergentWaypointName, routeModel) {
+        const indexOfDivergentLeg = this._findIndexOfLegContainingWaypointName(divergentWaypointName);
         const divergentLeg = this._legCollection[indexOfDivergentLeg];
 
         if (divergentLeg.isAirwayLeg) {
-            return this._appendRouteModelOutOfAirwayLeg(waypointName, routeModel);
+            return this._appendRouteModelOutOfAirwayLeg(divergentWaypointName, routeModel);
         }
 
         if (divergentLeg.isDirectLeg) {
-            return this._appendRouteModelOutOfDirectLeg(waypointName, routeModel);
+            return this._appendRouteModelOutOfDirectLeg(divergentWaypointName, routeModel);
         }
 
         if (divergentLeg.isSidLeg) {
-            return this._appendRouteModelOutOfSidLeg(waypointName, routeModel);
+            return this._appendRouteModelOutOfSidLeg(divergentWaypointName, routeModel);
         }
 
         if (divergentLeg.isStarLeg) {
-            return this._appendRouteModelOutOfStarLeg(waypointName, routeModel);
+            return this._appendRouteModelOutOfStarLeg(divergentWaypointName, routeModel);
         }
 
         throw new TypeError(`Expected known leg type, but received "${divergentLeg.legType}" ` +
             'type leg, preventing ability to determine the appropriate route merging strategy!'
         );
     }
 
+    /**
+     * Append a provided route model into this RouteModel when the divergent waypoint is in an airway leg
+     *
+     * This should only ever be called by `._appendRouteModelBeginningAtWaypointName()`
+     *
+     * @for RouteModel
+     * @method _appendRouteModelOutOfAirwayLeg
+     * @param divergentWaypointName {string} name of waypoint at which the two routes have continuity
+     * @param routeModel {RouteModel} the RouteModel to be absorbed into this
+     * @return {array} [success of operation, readback]
+     */
     _appendRouteModelOutOfAirwayLeg(divergentWaypointName, routeModel) {
         const indexOfDivergentLeg = this._findIndexOfLegContainingWaypointName(divergentWaypointName);
         const amendedAirwayLeg = this._createAmendedAirwayLegUsingDifferentExitName(
```

Comment:
a few lines down, I see `routeModel._legCollection`.  if `_legCollection` is intended to be private, it should not be accessed from outside the `routeModel`.  either make it public, or add a public getter for it.

Reply:
Apparently I pulled this slimy trick more times than I realized. Tracked them all down, and replaced with the new (tested) getter, `RouteModel.legCollection`.

done.